doc/reference/configuration/configuration_reference.rst

```diff
@@ -3599,6 +3599,290 @@ The ``replication`` section defines configuration parameters related to :ref:`re
     | Default: ``false``
     | Environment variable: TT_REPLICATION_ANON
 
+.. _configuration_reference_replication_autoexpel:
+
+.. confval:: replication.autoexpel
+
+    **Since:** :doc:`3.3.0 </release/3.3.0>`
+
+    The ``replication.autoexpel`` option designed for managing dynamic clusters using YAML-based configurations.
+    It enables the automatic expulsion of instances that are removed from the YAML configuration.
+
+    Only instances with names that match the specified prefix are considered for expulsion; all others are excluded.
+    Additionally, instances without a persistent name are ignored.
+
+    If an instance is in read-write mode and has the latest database schema, it initiates the expulsion of instances that:
+
+	- Match the specified prefix
+	- Absent from the updated YAML configuration
+
+    The expulsion process follows the standard procedure, involving the removal of the instance from the ``_cluster`` system space.
+
+    The ``autoexpel`` logic is activated during specific events:
+
+	- **Startup**. When the cluster starts, ``autoexpel`` checks and removes instances not matching the updated configuration.
+	- **Reconfiguration**. When the YAML configuration is reloaded, ``autoexpel`` compares the current state to the updated configuration and performs necessary expulsions.
+	- ``box.status`` **watcher event**. Changes detected by the ``box.status`` watcher also trigger the ``autoexpel`` mechanism.
+
+
+    ``autoexpel`` does not take any actions on newly joined instances unless one of the triggering events occurs.
+    This means that an instance meeting the ``autoexpel`` criterion can still join the cluster, but it may be removed
+    later during reconfiguration or on subsequent triggering events.
+
+    ..  NOTE::
+        The ``replication.autoexpel`` option governs the expelling process and is configurable at the replicaset, group, and
+        global levels. It is not applicable at the instance level.
+
+
+    Configuration fields
+
+    - ``by`` (string, default: ``nil``): specifies the ``autoexpel`` criterion. Currently, only ``prefix`` is supported and must be explicitly set.
+
+    - ``enabled`` (boolean, default: ``false``): enables or disables the ``autoexpel`` logic.
+
+    - ``prefix`` (string, default: ``nil``): defines the pattern for instance names that are considered part of the cluster.
+
+
+
+replication.autoexpel_by.*
+~~~~~~~~~~~~~
+
+    ``replication.autoexpel_by`` purpose is to define the criterion used for determining which instances in a cluster are
+    subject to the ``autoexpel`` process.
+
+    The ``by`` field helps differentiate between:
+
+    - Instances that are part of the cluster and should adhere to the YAML configuration.
+
+	- Instances or tools (e.g., CDC tools) that use the replication channel but are not part of the cluster configuration.
```

Comment:
The indentation seems broken (and also rendered unusual on the website).

Reply:
fixed